### PR TITLE
Refactoring particleset creation method

### PR DESCRIPTION
**Pull request by michaeldenes**
This PR is to change the particleset creation method, splitting it into two parts. One method allows users to create a particle set using one of the predefined initialisation maps, and the other method allows users to pass in their own release locations and plastic amounts.

### CODE REVIEW

**Comment by michaeldenes** [comment]
> This branch is now set to merge into `master`, but also includes `adding_new_tests`. Should it not merge into that second branch? Or should `adding_new_tests` not first be merged into `master`?

I want to PR `adding_new_tests` first, then this one. I merged `adding_new_tests` into this branch so I could run the pytests. So we can merge that first, then this one.

**Comment by erikvansebille** [comment]
> > This branch is now set to merge into `master`, but also includes `adding_new_tests`. Should it not merge into that second branch? Or should `adding_new_tests` not first be merged into `master`?
> 
> I want to PR `adding_new_tests` first, then this one. I merged `adding_new_tests` into this branch so I could run the pytests. So we can merge that first, then this one.

Yes, that would help me with reviewing, because now I will need to also go through all the changes of the `adding_new_tests` branch in https://github.com/OceanParcels/plasticparcels/pull/39/files

File: plasticparcels/constructors.py [hunk 1]
```diff
@@ -202,63 +202,32 @@ def create_fieldset(settings):
...
-                         'plastic_amount': particle_locations[release_quantity_name]}
-
     # Set the longitude, latitude, and plastic amount per particle
     lons = release_locations['lons']
     lats = release_locations['lats']
-    plastic_amounts = release_locations['plastic_amount']
+    if 'plastic_amount' in release_locations.keys():
+        plastic_amounts = release_locations['plastic_amount']
+    else:
+        plastic_amounts = np.zeros_like(lons)
```
Comment (erikvansebille) [inline plasticparcels/constructors.py:229]: Would setting these plastic_amounts to `np.nan` not make more sense?

Comment (michaeldenes) [reply plasticparcels/constructors.py:229]: I agree, that makes more sense (a zero might be assumed a true zero or a missing value, a NaN can only be interpreted as a missing value)

Comment (michaeldenes) [reply plasticparcels/constructors.py]: I agree, that makes more sense (a zero might be assumed a true zero or a missing value, a NaN can only be interpreted as a missing value). Have updated now.

**Pull request merged**